pandas/tests/plotting/test_frame.py

```diff
@@ -3419,6 +3419,23 @@ def test_xlabel_ylabel_dataframe_subplots(
         assert all(ax.get_ylabel() == str(new_label) for ax in axes)
         assert all(ax.get_xlabel() == str(new_label) for ax in axes)
 
+    def test_nullable_int_plot(self):
+        # GH 32073
+        dates = ["2008", "2009", None, "2011", "2012"]
+        df = pd.DataFrame(
+            {
+                "A": [1, 2, 3, 4, 5],
+                "B": [7, 5, np.nan, 3, 2],
+                "C": pd.to_datetime(dates, format="%Y"),
+                "D": pd.to_datetime(dates, format="%Y", utc=True),
```

Comment:
@MarcoGorelli This is the test case that hits the `elif is_datetime64tz_dtype(values.dtype)` branch in `safe_convert_to_ndarray`.

Reply:
@cvanweelden thanks for pointing me towards this.

Are you sure? I just tried running it and didn't hit it. Perhaps it was true before you changed
```python
safe_convert_to_ndarray(numeric_data[col])
```
to
```python
safe_convert_to_ndarray(numeric_data[col].values)
```
?

Reply:
Ah yeah, you are right. I wasn't aware that the underlying `values` were actually an ndarray with non-extended dtype, so I didn't know this change would have that effect. Thanks!